### ISSUE TITLE
Reimplement Admin in new fairseq but get bad valid loss

### ISSUE THREAD

**Issue by moonscar** [LiyuanLucasLiu/Transformer-Clinic#20]
We are trying reimplement admin-init method on fairseq v0.10.2, but get bad valid loss on wmt14 en-de data set.

training log below:
```
2021-04-07 20:45:11 | INFO | fairseq.trainer | begin training epoch 1  
2021-04-07 20:45:15 | INFO | fairseq.trainer | NOTE: overflow detected, setting loss scale to: 64.0  
2021-04-07 20:45:49 | INFO | fairseq.trainer | NOTE: overflow detected, setting loss scale to: 32.0  
2021-04-07 20:50:04 | INFO | fairseq.trainer | NOTE: overflow detected, setting loss scale to: 16.0  
2021-04-07 20:53:49 | INFO | train_inner | epoch 001:    503 / 6441 loss=11.988, nll_loss=11.486, ppl=2867.43, wps=20890.9, ups=0.97, wpb=21436.2, bsz=718.9, num_updates=500, lr=0.0001, gnorm=1.104, loss_scale=16, train_wall=512, wall=0  
2021-04-07 21:02:17 | INFO | train_inner | epoch 001:   1003 / 6441 loss=9.62, nll_loss=8.737, ppl=426.56, wps=20959, ups=0.98, wpb=21318.1, bsz=702.2, num_updates=1000, lr=0.0002, gnorm=0.794, loss_scale=16, train_wall=508, wall=0  
2021-04-07 21:10:46 | INFO | train_inner | epoch 001:   1503 / 6441 loss=8.377, nll_loss=7.298, ppl=157.34, wps=21125.3, ups=0.98, wpb=21509.4, bsz=693.8, num_updates=1500, lr=0.0003, gnorm=0.777, loss_scale=16, train_wall=508, wall=0  
2021-04-07 21:19:16 | INFO | train_inner | epoch 001:   2003 / 6441 loss=7.383, nll_loss=6.147, ppl=70.85, wps=21100, ups=0.98, wpb=21504.9, bsz=715.1, num_updates=2000, lr=0.0004, gnorm=0.738, loss_scale=16, train_wall=509, wall=0  
2021-04-07 21:27:45 | INFO | train_inner | epoch 001:   2503 / 6441 loss=6.627, nll_loss=5.27, ppl=38.57, wps=20886.9, ups=0.98, wpb=21251.9, bsz=684.3, num_updates=2500, lr=0.0005, gnorm=0.676, loss_scale=16, train_wall=508, wall=0  
2021-04-07 21:36:13 | INFO | train_inner | epoch 001:   3003 / 6441 loss=6.084, nll_loss=4.646, ppl=25.04, wps=21038.3, ups=0.98, wpb=21391, bsz=703.7, num_updates=3000, lr=0.0006, gnorm=0.568, loss_scale=16, train_wall=507, wall=0  
2021-04-07 21:42:27 | INFO | fairseq.trainer | NOTE: overflow detected, setting loss scale to: 8.0  
2021-04-07 21:44:43 | INFO | train_inner | epoch 001:   3504 / 6441 loss=5.803, nll_loss=4.329, ppl=20.1, wps=20929.2, ups=0.98, wpb=21350.7, bsz=701.5, num_updates=3500, lr=0.0007, gnorm=0.515, loss_scale=8, train_wall=509, wall=0  
2021-04-07 21:53:14 | INFO | train_inner | epoch 001:   4004 / 6441 loss=5.618, nll_loss=4.124, ppl=17.44, wps=21065, ups=0.98, wpb=21504.8, bsz=720.8, num_updates=4000, lr=0.0008, gnorm=0.473, loss_scale=8, train_wall=510, wall=0  
2021-04-07 22:01:43 | INFO | train_inner | epoch 001:   4504 / 6441 loss=5.466, nll_loss=3.957, ppl=15.53, wps=21172.3, ups=0.98, wpb=21562.4, bsz=706.3, num_updates=4500, lr=0.0009, gnorm=0.454, loss_scale=8, train_wall=508, wall=0  
2021-04-07 22:10:12 | INFO | train_inner | epoch 001:   5004 / 6441 loss=5.375, nll_loss=3.858, ppl=14.5, wps=21058.3, ups=0.98, wpb=21454, bsz=715.1, num_updates=5000, lr=0.001, gnorm=0.458, loss_scale=8, train_wall=509, wall=0  
2021-04-07 22:18:41 | INFO | train_inner | epoch 001:   5504 / 6441 loss=5.297, nll_loss=3.773, ppl=13.68, wps=20938.1, ups=0.98, wpb=21318.6, bsz=700.8, num_updates=5500, lr=0.000953463, gnorm=0.451, loss_scale=8, train_wall=508, wall=0  
2021-04-07 22:27:11 | INFO | train_inner | epoch 001:   6004 / 6441 loss=5.196, nll_loss=3.663, ppl=12.67, wps=21114.5, ups=0.98, wpb=21543.6, bsz=682.4, num_updates=6000, lr=0.000912871, gnorm=0.43, loss_scale=8, train_wall=509, wall=0  
2021-04-07 22:34:36 | INFO | fairseq_cli.train | begin validation on "valid" subset  
2021-04-07 22:34:39 | INFO | valid | epoch 001 | valid on 'valid' subset | loss 12.118 | nll_loss 11.627 | ppl 3161.8 | wps 76336 | wpb 13900.7 | bsz 500 | num_updates 6437  
2021-04-07 22:34:39 | INFO | fairseq_cli.train | begin save checkpoint  
2021-04-07 22:35:36 | INFO | fairseq.checkpoint_utils | saved checkpoint /nas/infrastructure/logs//test/checkpoint_best.pt (epoch 1 @ 6437 updates, score 12.118) (writing took 57.08395667281002 seconds)  
2021-04-07 22:35:36 | INFO | fairseq_cli.train | end of epoch 1 (average epoch stats below)  
2021-04-07 22:35:36 | INFO | train | epoch 001 | loss 6.782 | nll_loss 5.47 | ppl 44.33 | wps 20831.7 | ups 0.97 | wpb 21426 | bsz 702.5 | num_updates 6437 | lr 0.00088134 | gnorm 0.607 | loss_scale 8 | train_wall 6551 | wall 0
```

init value
```
enc 0 1.0                                                                                                                                                                                                                                                                     
enc 0 1.3212890625                                                                                                                                                                                                                                                            
enc 1 1.4443359375                                                                                                                                                                                                                                                            
enc 1 1.4873046875                                                                                                                                                                                                                                                            
enc 2 1.6025390625                                                                                                                                                                                                                                                            
enc 2 1.6533203125                                                                                                                                                                                                                                                            
enc 3 1.7646484375                                                                                                                                                                                                                                                            
enc 3 1.828125                                                                                                                                                                                                                                                                
enc 4 1.9287109375                                                                                                                                                                                                                                                            
enc 4 1.9892578125                                                                                                                                                                                                                                                            
enc 5 2.083984375                                                                                                                                                                                                                                                             
enc 5 2.13671875                                                                                                                                                                                                                                                              
enc 6 2.21484375                                                                                                                                                                                                                                                              
enc 6 2.271484375                                                                                                                                                                                                                                                             
enc 7 2.34765625                                                                                                                                                                                                                                                              
enc 7 2.40625                                                                                                                                                                                                                                                                 
enc 8 2.482421875                                                                                                                      
enc 8 2.53515625                                                                                                                       
enc 9 2.6015625                                                                                                                                                                                                                                                               
enc 9 2.65625                                                                                                                                                                                                                                                                 
enc 10 2.728515625                                                                                                                                                                                                                                                            
enc 10 2.78515625                                                                                                                                                                                                                                                             
enc 11 2.849609375                                                                                                                     
enc 11 2.90625                                                                                                                                                                                                                                                                
dec 0 1.0                                                                                                                                                                                                                                                                     
dec 0 1.3544921875                                                                                                                                                                                                                                                            
dec 0 1.45703125                                                                                                                                                                                                                                                              
dec 1 1.5732421875                                                                                                                                                                                                                                                            
dec 1 1.646484375                                                                                                                                                                                                                                                             
dec 1 1.7333984375                                                                                                                                                                                                                                                            
dec 2 1.83203125                                                                                                                                                                                                                                                              
dec 2 1.908203125                                                                                                                                                                                                                                                             
dec 2 2.005859375                                                                                                                                                                                                                                                             
dec 3 2.09375                                                                                                                                                                                                                                                                 
dec 3 2.16796875                                                                                                                                                                                                                                                              
dec 3 2.234375                                                                                                                                                                                                                                                                
dec 4 2.3125                                                                                                                                                                                                                                                                  
dec 4 2.3828125                                                                                                                                                                                                                                                               
dec 4 2.447265625                                                                                                                                                                                                                                                             
dec 5 2.525390625                                                                                                                      
dec 5 2.60546875                                                                                                                       
dec 5 2.66015625                                                   
dec 6 2.728515625                                                  
dec 6 2.798828125                                                  
dec 6 2.84765625                                                   
dec 7 2.91015625                                                   
dec 7 2.970703125                                                  
dec 7 3.02734375                                                   
dec 8 3.083984375                                                  
dec 8 3.140625                                                     
dec 8 3.19140625                                                   
dec 9 3.2421875                                                    
dec 9 3.29296875                                                   
dec 9 3.3359375                                                    
dec 10 3.390625                                                    
dec 10 3.447265625                                                 
dec 10 3.494140625                                                 
dec 11 3.546875                                                    
dec 11 3.59765625                                                  
dec 11 3.650390625  
```

training args
`
Namespace(activation_dropout=0.0, activation_fn='relu', adam_betas='(0.9, 0.98)', adam_eps=1e-08, adaptive_input=False, adaptive_softmax_cutoff=None, adaptive_softmax_dropout=0, all_gather_list_size=16384, arch='transformer_wmt_en_de', attention_dropout=0.0, batch_size=None, batch_size_valid=None, best_checkpoint_metric='loss', bf16=False, bpe=None, broadcast_buffers=False, bucket_cap_mb=25, checkpoint_shard_count=1, checkpoint_suffix='', clip_norm=0.0, cpu=False, criterion='label_smoothed_cross_entropy', cross_self_attention=False, curriculum=0, data='/nas/infrastructure/stand-translation-data/data-bin/wmt14_en_de_joined_dict', data_buffer_size=10, dataset_impl=None, ddp_backend='c10d', decoder_attention_heads=8, decoder_embed_dim=512, decoder_embed_path=None, decoder_ffn_embed_dim=2048, decoder_input_dim=512, decoder_layerdrop=0, decoder_layers=12, decoder_layers_to_keep=None, decoder_learned_pos=False, decoder_normalize_before=False, decoder_output_dim=512, device_id=0, disable_validation=False, distributed_backend='nccl', distributed_init_method='tcp://localhost:16144', distributed_no_spawn=False, distributed_num_procs=4, distributed_port=-1, distributed_rank=0, distributed_world_size=4, distributed_wrapper='DDP', dropout=0.1, empty_cache_freq=0, enable_postln_stable=True, encoder_attention_heads=8, encoder_embed_dim=512, encoder_embed_path=None, encoder_ffn_embed_dim=2048, encoder_layerdrop=0, encoder_layers=12, encoder_layers_to_keep=None, encoder_learned_pos=False, encoder_normalize_before=False, eval_bleu=False, eval_bleu_args=None, eval_bleu_detok='space', eval_bleu_detok_args=None, eval_bleu_print_samples=False, eval_bleu_remove_bpe=None, eval_tokenized_bleu=False, fast_stat_sync=False, find_unused_parameters=False, finetune_from_model=None, fix_batches_to_gpus=False, fixed_validation_seed=None, fp16=True, fp16_init_scale=128, fp16_no_flatten_grads=False, fp16_scale_tolerance=0.0, fp16_scale_window=None, gen_subset='test', ignore_prefix_size=0, keep_best_checkpoints=-1, keep_interval_updates=-1, keep_last_epochs=-1, label_smoothing=0.1, layernorm_embedding=False, left_pad_source='True', left_pad_target='False', load_alignments=False, localsgd_frequency=3, log_format=None, log_interval=500, lr=[0.001], lr_scheduler='inverse_sqrt', max_epoch=30, max_source_positions=1024, max_target_positions=1024, max_tokens=6000, max_tokens_valid=6000, max_update=0, maximize_best_checkpoint_metric=False, memory_efficient_bf16=False, memory_efficient_fp16=False, min_loss_scale=0.0001, min_lr=-1.0, model_parallel_size=1, no_cross_attention=False, no_epoch_checkpoints=True, no_last_checkpoints=False, no_progress_bar=True, no_save=False, no_save_optimizer_state=False, no_scale_embedding=False, no_seed_provided=False, no_token_positional_embeddings=False, nprocs_per_node=4, num_batch_buckets=0, num_shards=1, num_workers=1, optimizer='adam', optimizer_overrides='{}', patience=-1, pipeline_balance=None, pipeline_checkpoint='never', pipeline_chunks=0, pipeline_decoder_balance=None, pipeline_decoder_devices=None, pipeline_devices=None, pipeline_encoder_balance=None, pipeline_encoder_devices=None, pipeline_model_parallel=False, profile=False, quant_noise_pq=0, quant_noise_pq_block_size=8, quant_noise_scalar=0, quantization_config_path=None, report_accuracy=False, required_batch_size_multiple=8, required_seq_len_multiple=1, reset_dataloader=False, reset_lr_scheduler=False, reset_meters=False, reset_optimizer=False, restore_file='admin.pt', save_dir='/nas/infrastructure/logs//test', save_interval=1, save_interval_updates=0, scoring='bleu', seed=1, sentence_avg=False, shard_id=0, share_all_embeddings=False, share_decoder_input_output_embed=True, skip_invalid_size_inputs_valid_test=False, slowmo_algorithm='LocalSGD', slowmo_momentum=None, source_lang='en', stop_time_hours=0, target_lang='de', task='translation', tensorboard_logdir=None, threshold_loss_scale=None, tie_adaptive_weights=False, tokenizer=None, tpu=False, train_subset='train', truncate_source=False, update_freq=[1], upsample_primary=1, use_bmuf=False, use_old_adam=False, user_dir=None, valid_subset='valid', validate_after_updates=0, validate_interval=1, validate_interval_updates=0, warmup_init_lr=-1, warmup_updates=5000, weight_decay=0.0001, zero_sharding='none')
`

Any advice for me ?
